rfcs/0088-commit-message-guideline.md

```diff
@@ -0,0 +1,187 @@
+commit messages can affect negatively the code quality of future changes that
+would otherwise benefit from past good commit messages if they existed.
+
+Beyond code itself, poorly written commit messages can also affect the community
+in other ways. For example, by not providing to new contributors a consistent
+and complete history or context for the code changes, it can work as a barrier
+for new contributions because much more time will be necessary trying to
+understand what motivated a past critical but unclear change.
+
+Hence this Commit Message Guideline can help contributors to write good commit
+messages and so improve the current situation regarding the TVM commit logs.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Commit Message Guideline
+
+Apache TVM uses the Github (GH) platform for patch submission and code review
+via Pull Requests (PRs). The final commit (title and body) that is merged into
+the Apache TVM main tree is composed of the PR's title and body and must be kept
+updated and reflecting the new changes in the code as per the reviews and
+discussions.
+
+Although these guidelines apply essentially to the PRs’ title and body messages,
+because GH auto-generates the PR’s title and body from the commits on a given
+branch, it’s recommended to follow these guidelines right from the beginning,
+when preparing commits in general to be submitted to the Apache TVM project.
+This will ease the creation of a new PR, avoiding rework, and also will help the
+review.
+
+The rules below will help to achieve uniformity that has several benefits, both
+for review and for the code base maintenance as a whole, helping you to write
+commit messages with a good quality suitable for the Apache TVM project,
+allowing fast log searches, bisecting, and so on.
+
+_PR/commit title_:
+
+* Guarantee a title exists (enforced);
+* Don’t use Github usernames in the title, like @username (enforced);
+* Check if a tag should be present as a hint about what component(s) of the code
+  the commits “touch”. For example [BugFix], [CI], [microTVM], and [TVMC]. Tags
+  go between square brackets and appear first in the title. If more than one tag
+  exist, multiple brackets should be used, like [BugFix][CI]. The case
+  recommended for tags, in geral, is the upper camel case. For example, prefer
+  the forms [Fix], [BugFix], and [Docker] instead of [fix], [bug_fix], and
+  [docker]. Acronyms should be kept as such so, for example, use [CI] and [TVMC]
+  instead of [ci] and [tvmc]. Tags help reviewers to identify the PRs they
+  can/want to review and also help the release folks when generating the release
+  notes (enforced);
+* Use an imperative mood. Avoid titles like “Added operator X” and “Updated
+  image Y in the CI”, instead use the forms “Add feature X” and “Update image Y
+  in the CI” instead;
+* Observe proper use of caps at the beginning (uppercase for the first letter)
```

Comment:
Just to clarify, we only use uppercase for the first letter and acronyms right? This means:
```
Fix TVM use of OpenCL library
```
is better than
```
Fix TVM Use of OpenCL Library
```

Reply:
The example reflects a few variants, do you want us to include the particular one you suggest?

Reply:
I don't actually have a strong opinion :-)  Let's just clarify it a little bit so that readers could better follow

Reply:
@junrushao1994 That's right, i.e. form `Fix TVM use of OpenCL library` is better than form `Fix TVM Use of OpenCL Library`. But since in the RFC we used exactly the best form, i.e. `Fix TVM use of OpenCL library`, as the example I'm not sure how we could clarify further ... but wait, hm I think I get it, what can be ambiguous is that when it says "uppercase for the first letter" it can also be interpreted as the first letter of each word is capitalized, i.e. it can be interpreted as title case (https://en.wikipedia.org/wiki/Title_case), right? If so we should clarify saying explicitly it's uppercase for the first letter in the title and that title case must be avoided? Am I following you correctly?

Reply:
@junrushao friendly ping :)